Mathlib/Algebra/BigOperators/Ring/Nat.lean

```diff
@@ -0,0 +1,35 @@
+/-
+Copyright (c) 2024 Pim Otte. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Pim Otte
+-/
+import Mathlib.Algebra.BigOperators.Group.Finset
+import Mathlib.Algebra.Ring.Parity
+
+/-!
+# Big operators on a finset in the natural numbers
+
+This file contains the results concerning the interaction of finset big operators with natural
+numbers.
+-/
+
+variable {ι : Type*}
+
+namespace Finset
+
+lemma even_sum_of_even_card_odd {s : Finset ι} (f : ι → ℕ)
```

Comment:
It would be nice to turn this into an `iff`

Reply:
and to have an `iff` with 2 evens replaced by odd.

Reply:
Can probably be golfed further, but here is a slicker proof:
```lean
lemma even_sum_iff_even_card_odd {s : Finset ι} (f : ι → ℕ) :
    Even (∑ i ∈ s, f i) ↔ Even (s.filter fun x ↦ Odd (f x)).card := by
  rw [← Finset.sum_filter_add_sum_filter_not _ (fun x ↦ Even (f x)), Nat.even_add]
  simp only [Finset.mem_filter, and_imp, imp_self, implies_true, Finset.even_sum, true_iff]
  rw [Nat.even_iff, Finset.sum_nat_mod, Finset.sum_filter]
  simp (config := { contextual := true }) only [← Nat.odd_iff_not_even, Nat.odd_iff.mp]
  simp_rw [← Finset.sum_filter, ← Nat.even_iff, Finset.card_eq_sum_ones]
```

Reply:
Thanks, used this and added the Odd variant